engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -915,9 +919,50 @@ private void applyProperties(final ComponentBuilder builder,
                     final Converter<?> customConverter = configuredProperty.createCustomConverter();
                     final Object value = stringConverter.deserialize(stringValue, configuredProperty.getType(),
                             customConverter);
+               
+                    if (name.equals(CoalesceMultipleFieldsTransformer.PROPERTY_UNITS)) {
+                        /* This part of the code refers to the situation when we
+                         open the job as a template where we need to replace
+                         the name of the columns with the mapped ones */
+                        final CoalesceUnit[] units = (CoalesceUnit[]) value;
+
+                        final ArrayList<CoalesceUnit> newUnitsList = new ArrayList<CoalesceUnit>();
+                        final Set<Entry<String, InputColumn<?>>> mappingColumnsSet = mappingInputColumns.entrySet();
+                        for (int i = 0; i < units.length; i++) {
+                            final String[] oldInputColumnNames = units[i].getInputColumnNames();
+                            final ArrayList<InputColumn<?>> newInputColumns = new ArrayList<InputColumn<?>>();
+                            for (Entry<String, InputColumn<?>> entry : mappingColumnsSet) {
+                                final String column_id = entry.getKey();
+                                final String path = getPath(columnsTypes, column_id);
+                                for (int j = 0; j < oldInputColumnNames.length; j++) {
+                                    /* Eg. <column id="col_given_name" path="given_name" type="STRING"/>
+                                     The path is found in the name of the column: 
+                                    datastores.customers.csv.given_name */
+                                    if (oldInputColumnNames[j].contains(path)) {
+                                        // add the mapped column.  
+                                        newInputColumns.add(entry.getValue());
+                                    }
+                                }
+                            }
+                            //create new coalesce unit with the mapped columns. 
+                            if (!newInputColumns.isEmpty()) {
+                                final CoalesceUnit newCoalesceUnit = new CoalesceUnit(newInputColumns);
+                                newUnitsList.add(newCoalesceUnit);
+                            }
+                        }
 
+                        // If in the coalesce units we have inputcolumns names and not the physical names then we keep
+                        // the original value. Eg value="[&amp;#91;EQ name&amp;#44;NEQ name&amp;#93;]"/>
+                        if (!newUnitsList.isEmpty()) {
```

Comment:
:+1: (It might be a little easier to follow conversations is you instead reply directly to the original comment with something like "that's handled in line xxx", otherwise thing get very messy very fast if there are multiple threads of discussion).